calico/felix/profilerules.py

```diff
@@ -104,6 +104,11 @@ def on_rules_update(self, profile_id, profile, force_reprogram=False):
                        profile_id)
             # Pass in_sync=True because we now explicitly know this profile is
             # in sync, even if this is a deletion.
+            #PLW: I can see we could do this - is it worth doing? I worry that
+            # we might program a profile for which the tags are not in place yet
+            # (say). If we do so, are we actually helping at all? Removing
+            # this bit of logic would make things slightly simpler, and
+            # not sure I see the benefit.
```

Comment:
I _think_ so.  If the profile doesn't use tags then it's better.  It also allows us to handle profile deletions during the GR window.